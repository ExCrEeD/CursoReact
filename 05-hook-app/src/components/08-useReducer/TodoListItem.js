import React from 'react'
import "./style.css"

export const TodoListItem = ({task,index,handleDelete,handleToggle}) => {
    return (
        <li
            className="list-group-item"
        >
            <p
                className={`${task.done && 'complete'}`}
                onClick={() => handleToggle(task.id)}
            >{index + 1}.{task.desc}</p>
            <button
                className="btn btn-danger"
                onClick={() => handleDelete(task.id)}
            >borrar</button>
        </li>
    )
}
