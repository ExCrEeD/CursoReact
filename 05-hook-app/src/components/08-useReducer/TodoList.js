import React from 'react'
import { TodoListItem } from './TodoListItem'
import "./style.css"

export const TodoList = ({todos,handleDelete,handleToggle}) => {
    return (
        <ul className='list-group list-group-flush'>
            {todos.map((task,index)=>
                    <TodoListItem 
                        key={task.id}
                        task={task}
                        index={index}
                        handleDelete={handleDelete}
                        handleToggle={handleToggle}
                        />
            )}
        </ul>
    )
}
