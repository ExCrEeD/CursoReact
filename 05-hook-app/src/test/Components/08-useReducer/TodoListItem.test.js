import shallow from "enzyme/build/shallow"
import { TodoListItem } from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe('pruebas en <TodoListItem/>', () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(<TodoListItem
        task = {demoTodos[0]}
        index = {0}
        handleDelete = {handleDelete}
        handleToggle = {handleToggle}
    />);


    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de llamar la funcion handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
    });

    test('debe de llmar la funcion handleToggle', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
    });

    test('debe de mostrar el texto correctamente', () => {
         const p = wrapper.find('p');
         expect(p.text().trim()).toBe(`1.${demoTodos[0].desc}`);
    });
        
    test('debe de tener la clase complete', () => {

        const todo = demoTodos[0];
        todo.done = true;
        const wrapper = shallow(<TodoListItem
            task = {todo}
            index = {0}
            handleDelete = {handleDelete}
            handleToggle = {handleToggle}
        />);
        const p = wrapper.find('p');
        expect(p.hasClass('complete')).toBe(true);
   });
       
})
