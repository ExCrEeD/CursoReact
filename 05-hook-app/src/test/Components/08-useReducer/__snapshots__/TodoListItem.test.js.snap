// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoListItem/> debe mostrarse correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    .
    aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    borrar
  </button>
</li>
`;
